tensorflow/compiler/xla/service/hlo_tfgraph_builder.cc

```diff
@@ -162,7 +162,7 @@ void HloTfGraphBuilder::SetNodeAttrs(const HloInstruction* instruction,
       // For tuples, emit the full shape because the layout of a tuple is not
       // represented in a single Layout field.
       layout_string = ShapeUtil::HumanStringWithLayout(instruction->shape());
-    } else {
+    } else if (instruction->shape().has_layout()) {
```

Comment:
Will it be obvious to readers why we have to do this check here?

Reply:
Done :)